proxy/proxy.go

```diff
@@ -958,7 +958,14 @@ func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Co
 			p.tracing.setTag(ctx.proxySpan, HTTPStatusCodeTag, uint16(status))
 			return nil, &proxyError{err: err, code: status}
 		}
-
+		if req.URL.Scheme == "fastcgi" {
```

Comment:
It should be possible to handle this case in fastcgi roundtripper internally and respond with normal 404 instead of returning a error. This way there would be no need to this and proxy error special handling.

Reply:
yes, makes sense. Done.